packages/labs/analyzer/src/lib/javascript/classes.ts

```diff
@@ -119,48 +122,45 @@ export const getClassMembers = (
       //
       // This is ok for now because these are rare ways to "declare" a field,
       // especially in web components where you shouldn't have constructor parameters.
-      node.body.statements.forEach((node) => {
+      member.body.statements.forEach((statement) => {
         if (
-          typescript.isExpressionStatement(node) &&
-          typescript.isBinaryExpression(node.expression) &&
-          node.expression.operatorToken.kind ===
-            typescript.SyntaxKind.EqualsToken &&
-          typescript.isPropertyAccessExpression(node.expression.left) &&
-          node.expression.left.expression.kind ===
-            typescript.SyntaxKind.ThisKeyword
+          typescript.isExpressionStatement(statement) &&
+          isConstructorFieldInitializer(statement.expression, typescript)
         ) {
-          const name = node.expression.left.name.getText();
+          const name = statement.expression.left.name.getText();
```

Comment:
here and elsewhere: getText() is almost always a mistake to call outside of debug statements, because it exists on every node but doesn't have a reasonable value for a lot of them (and when called on artificial nodes that weren't in the original source code it may throw an error)

Reply:
This is already used in a bunch of places. I can file an issue to change them.

Reply:
Yeah, we should honestly add a lint rule to catch and prevent calls to getText(). We did that on the internal version of grimlock and it prevented a bunch of issues.

Andrew actually ran into an issue _today_ where .getText() did the wrong thing (when called on a SourceFile it doesn't include all the content of the file (!), it misses leading comments. SourceFile has a .text property with the full text of the file)

Reply:
![image](https://github.com/lit/lit/assets/15080861/b56f22af-28b4-4be9-995b-95ef2d030f79)